lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -656,14 +656,8 @@ def _get_json_from_installed_dir(
     try:
         with open(b_json_filepath, 'rb') as manifest_fd:
             b_json_text = manifest_fd.read()
-    except (IOError, OSError):
-        raise LookupError(
-            "The collection {manifest!s} path '{path!s}' does not exist.".
-            format(
-                manifest=filename,
-                path=to_native(b_json_filepath),
-            )
-        )
+    except OSError as ex:
+        raise LookupError(f"The collection {filename!r} path {to_native(b_json_filepath)!r} does not exist.") from ex
```

Comment:
Should filename be an `!r` tho? It changes the semantics of the original message.
Also, sticking nested expressions into f-strings hurts readability due to added cognitive complexity inline. So I'd either keep that one in `str.format()` or extracted it into a variable.

Reply:
Also, the second `!r` might produce different quotes than the original single ones.

Reply:
It's usually better to let Python quote filenames instead of doing it manually, so `!r` makes sense here. Yes, the quoting may be different, but that shouldn't be an issue.

As for the `to_native`, it really should be removed (along with `to_text` and `to_bytes`)... but I'd save those changes for another PR, since it looks like a lot of the galaxy code makes use of passing around `bytes` instead of `str`. In the meantime, I don't see an issue with leaving it inline in the f-string. Hopefully it won't stick around too much longer.

Reply:
Ack.